### PR TITLE
new missing city segment checks

### PR DESCRIPTION
**Pull request by dalverson**
Also fix for report details in left panel and CSP fix.

### CODE REVIEW

File: src/basic.js [hunk 1]
```diff
@@ -271,6 +289,14 @@ function ForceHLAllObjects() {
 	HLAllObjects();
 }
 
+/**
+ * Delay then Force Highlight objects
+ *  This is used to wait for DOM to populate before filling in problem details in left panel.
+ */
+function delayForceHLAllObjects() {
+	setTimeout(ForceHLAllObjects, 100);
```
Comment (berestovskyy) [inline src/basic.js:297]: The delay looks fragile, maybe we should wait until the object appears? A quick search: https://stackoverflow.com/questions/5525071/how-to-wait-until-an-element-exists

File: src/validate.js [hunk 2]
```diff
@@ -2649,6 +2649,24 @@ function F_VALIDATE(disabledHL) {
 				}
 			}// global speed limit check
 
+			// GROUP isDrivable
+			// on named segment, make sure theres a city on primary or alt
+			if (!cityLen && streetLen && RT_RAMP !== roadType && RT_FREEWAY !== roadType) {
```
Comment (berestovskyy) [inline src/validate.js:2654]: nit: for the sake of performance, we should start with the cheapest checks first, so it seems `isLimitOk()` should be somewhere here, at the end of the condition.

**Comment by berestovskyy** [comment]
Looks good, thanks for your contribution!

File: src/basic.js (hunk 1 above)
Comment (dalverson) [reply src/basic.js:297]: yes, it is a little fragile. But this is being called on selectionchanged, so we dont know for certain what element we might need.

File: src/validate.js (hunk 2 above)
Comment (dalverson) [reply src/validate.js:2654]: Since 54 & 55 both use this, it would have to be (isLimitOk(54) || isLimitOk(55))

Comment (dalverson) [reply src/validate.js]: limit checks added on first condition

**Review by berestovskyy — APPROVED** [review]

**Pull request merged**